### DISCUSSION TITLE
Error: No Firebase App '[DEFAULT]' has been created - call firebase.initializeApp()

### DISCUSSION THREAD

**Discussion by Kamlesh-62** [invertase/react-native-firebase#6927] [General]
Hello, 

I am using @react-native-firebase/messaging for push notifications in react-native.  I don't know why I am getting this error after I have tried so many solutions still getting the same error again and again.

Error: No Firebase App '[DEFAULT]' has been created - call firebase.initializeApp() this is error.

I read the document very carefully it says that "there is no need to manually call the [initializeApp](https://firebase.google.com/docs/web/setup#add-sdks-initialize) method with your project credentials. The native Android & iOS SDKs automatically connect to your Firebase project using the credentials provided during the [Getting Started](https://rnfirebase.io/) installation steps." I have followed all steps for installation.

```
react-native info
System:
    OS: macOS 12.6
    CPU: (6) x64 Intel(R) Core(TM) i5-8600 CPU @ 3.10GHz
    Memory: 8.37 GB / 48.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.17.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.15.0 - /usr/local/bin/npm
    Watchman: 2023.02.13.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.3 AI-213.7172.25.2113.9123335
    Xcode: 14.2/14C18 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.17 - /usr/bin/javac
    Python: Not Found
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.9.0 => 16.9.0 
    react-native: ^0.63.3 => 0.63.5 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

This is what my code look like!

import messaging from '@react-native-firebase/messaging';

```
**useEffect(() => {
    const useEfectFunction = async () => {
      try{
      const responseFirebase = await `messaging().getToken();`
      console.log("responseFirebase=>", responseFirebase)
      messaging().onNotificationOpenedApp(remoteMessage => {
        if (remoteMessage) {
          console.log("onNotificationOpenedApp")
                 }
             });
      messaging().setBackgroundMessageHandler(async remoteMessage => {
        if (remoteMessage) {
          console.log("setBackgroundMessageHandler")
                 }
      });
           console.log("responseFirebase=>", responseFirebase)
      await messaging()
        .getInitialNotification()
        .then(remoteMessage => {
          if (remoteMessage) {
                     }
        });
    }catch (error) {
      console.log("erroror",error)
    }
  }
    useEfectFunction();
  }, []);**
```
package.json
```
 "@react-native-firebase/app": "^17.3.0",
    "@react-native-firebase/messaging": "^17.3.0",

```
Really appreciate your help

**Comment by mikehardy** [comment]
Hi there! This looks like an issue that has been filed as a discussion topic instead, which conveniently ignores the need/effort required to post certain important information to help you. Like, what platform? etc

Please triple-check you've followed all the install instructions: https://rnfirebase.io/#installation

Please examine this script which demonstrates a complete + correct integration starting from scratch, noting carefully for instance that on iOS platform you do need to alter the AppDelegate file https://github.com/mikehardy/rnfbdemo/blob/main/make-demo.sh

**Comment by HiImYong** [comment]
hi. did you solved it?

**Comment by davidcort** [comment]
Same issue here :(

**Reply by ahmedsweng** [reply]
Hello @davidcort @HiImYong @Kamlesh-62 to solve this problem you should first verify that you already add the specific plugin for each package you from @react-native-firebase into you app.json file like that:
```json
"plugins": [
      "expo-router",
      "@react-native-firebase/app",
      "@react-native-firebase/auth",
      "@react-native-google-signin/google-signin"
    ]
```

here I am using auth and app in your case add what you need and rebuild your app

**Reply by davidcort** [reply]
Sorry, it was a git issue because the specific config on Android, not been there. I realize the config of the documentation and works like a charm :)

**Comment by Pattygeek** [comment]
> Sorry, it was a git issue because the specific config on Android, not been there. I realize the config of the documentation and works like a charm :)

What config exactly did you use that worked?

**Reply by kyunkakata** [reply]
My problem is not add google-services in build.gradle.
You can follow the doc here:
https://rnfirebase.io/#configure-firebase-with-android-credentials
